### DISCUSSION TITLE
How to obtain AtomOr list?

### DISCUSSION THREAD

**Discussion by iwwwish** [rdkit/rdkit#5032] [Q&A] [answered]
Hi all,

I am trying to read in a structure with list atoms (see example below where Cl, Br are the individual atoms in the list [Cl,Br]). RDKit reads the structure and displays it well, but I'm unable to access/obtain the individual atoms. RDKit detects this as a query atom and there is a query description attached that says it is an `AtomOr` list and further lists the `AtomAtomicNum` for each atom. However, the `getQuery()` method does not work. According to the documentation, `getQuery()` method under `QueryAtom` class should return the current query but somehow the actual implementation in `Atom` class says that the method is NOT CALLABLE.

Is there any workaround for this (other than parsing the query description)? Thank you in advance!

<img width="994" alt="Screen Shot 2022-02-22 at 2 13 32 AM" src="https://user-images.githubusercontent.com/5487952/155081287-c1082cee-6248-42fb-87f7-8725b3274af8.png">

**Comment by greglandrum** [comment]
@iwwwish : What do you want to do with the results? There may be an easy way or it may involve parsing some text.

**Reply by iwwwish** [reply]
Hi Greg!

I'm enumerating all possible structures given a Markush structure with R1, R2, R3, etc. and a list of reagents available for each R group. Some of these Markush structures have list atoms. In such cases, I would like to have twice the number of molecules in the enumerated library (assuming there are only two list atoms) where each molecule has either of the list atoms.

Thank you!

**Comment by greglandrum** [comment, marked as answer]
Hi @iwwwish,
unfortunately we don't have anything to make that particular use case easy.
Probably your best bet is to use the atom's `GetSmarts()` method and the parse the output of that:
```
In [8]: m = Chem.MolFromSmarts('[C][C,n,#15]')

In [9]: print(m.GetAtomWithIdx(1).GetSmarts())
[C,n,#15]
```
I intentionally made that one a bit complicated to give a sense of the breadth of what can show up in "simple" atom list SMARTS.

**Reply by iwwwish** [reply]
Thank you, Greg! I think I'll do that.

**Comment by iwwwish** [comment]
I could write this up quickly and it works for time being.

```
def get_list_atoms(mol):
    atom_map = {}
    for a in mol.GetAtoms():
        if a.HasQuery():
            if a.GetIsotope() == 0: # this is a not yet a list of atoms
                sm = a.GetSmarts()
                sm = sm[1:-1] # stripping the square backets
                sm_atms = sm.split(",")
                if len(sm_atms) > 1: # now, this is a list of atoms
                    list_atms = []
                    for atm in sm_atms:
                        list_atm = Chem.AtomFromSmarts('[' + atm + ']')
                        list_atm.SetAtomMapNum(a.GetIdx()) # retaining the atom index
                        list_atms.append(list_atm)
                    if list_atms:
                        atom_map[a.GetIdx()] = list_atms
    return atom_map
```
<img width="992" alt="Screen Shot 2022-02-22 at 2 27 30 PM" src="https://user-images.githubusercontent.com/5487952/155204635-6938c6ae-8540-451d-88b9-949195cf7026.png">
